### ISSUE TITLE
Resetting ride let guest out in the air

### ISSUE THREAD

**Issue by telk5093** [OpenRCT2/OpenRCT2#10902]
**OS:** Windows 10
**Version:** 0.2.4
**Commit/Build:** 8737d3c

Guest are exited with incorrect position when resetting rides by double clicking red flag/sign.

**Steps to reproduce:**
1. Build any ride
2. Let guest ride it.
3. Reset it by double clicking red signal button/red flag button
4. Guests are spawned/exited in the air.

**Screenshots / Video:**
![2](https://user-images.githubusercontent.com/10726524/76208605-5d8bdf00-6243-11ea-82bb-3f9164b68495.gif)

**Comment by duncanspumpkin** [comment]
The old code looked to see if you had reached the correct tile we need to check the new code isn't looking for matching the coordinate as that is too accurate.

**Comment by AaronVanGeffen** [comment]
This bug can do worse as well. Over here, editing a boat hire just made all the guests aboard instantly drown just now…

**Comment by AaronVanGeffen** [comment]
Bisected this to 57a40c0657e8a7ed1aa2fc2721f5da3b12f9137a:

```
git bisect start
# bad: [e27a1a230a0c6e8b73c46dc04893068edb64837f] Fix regression in boat hire pathfinding (#10958)
git bisect bad e27a1a230a0c6e8b73c46dc04893068edb64837f
# good: [d5c4d78d493c0238742f196bbe2d04f3116e0c75] Move vehicle track location to CoordsXYZ (#10856)
git bisect good d5c4d78d493c0238742f196bbe2d04f3116e0c75
# good: [cded8313d782e8cc002514a60ade5ce8d7ae91dc] Replace many occurrences of LocationXY16
git bisect good cded8313d782e8cc002514a60ade5ce8d7ae91dc
# bad: [aef748af146496cb3b75ff7e6575164e46668f33] Merge pull request #10901 from Gymnasiast/refactor/guest-window-many-pages
git bisect bad aef748af146496cb3b75ff7e6575164e46668f33
# good: [2eb967b30d52f1c0720c4f995cf80f4c54aa848a] Change station start to CoordsXY
git bisect good 2eb967b30d52f1c0720c4f995cf80f4c54aa848a
# good: [d64b73c29a68f511a326a1ad9a48bfedac026d7c] Clean up RCT1/2 constants (#10855)
git bisect good d64b73c29a68f511a326a1ad9a48bfedac026d7c
# bad: [be13c1fc87eaee50c0425a843af6b89698cb8f59] Refactor more TileCoords
git bisect bad be13c1fc87eaee50c0425a843af6b89698cb8f59
# bad: [57a40c0657e8a7ed1aa2fc2721f5da3b12f9137a] Extend PathElement
git bisect bad 57a40c0657e8a7ed1aa2fc2721f5da3b12f9137a
# good: [383d5d99e1e0a835b6f336e09ab3b797942e6662] Change entry index size of Small and Large scenery elements
git bisect good 383d5d99e1e0a835b6f336e09ab3b797942e6662
# first bad commit: [57a40c0657e8a7ed1aa2fc2721f5da3b12f9137a] Extend PathElement
```

**Comment by AaronVanGeffen** [comment]
I believe this may be down to world/Footpath.cpp still using old flags rather than the one that have changed, e.g.:

https://github.com/OpenRCT2/OpenRCT2/blob/e27a1a230a0c6e8b73c46dc04893068edb64837f/src/openrct2/world/Footpath.cpp#L1318-L1332
